src/scaler/spreadscaler/link.rs

```diff
@@ -0,0 +1,193 @@
+                    && linkdef.link_name == self.config.provider_link_name =>
+            {
+                self.reconcile().await
+            }
+            _ => Ok(Vec::new()),
+        }
+    }
+
+    async fn reconcile(&self) -> Result<Vec<Command>> {
+        if let (Some(actor_id), Some(provider_id)) = (self.actor_id.get(), self.provider_id.get()) {
+            //TODO: check to see if linkdef exists. Might need to be deleted first if values differ
+            Ok(vec![Command::PutLinkdef(PutLinkdef {
+                actor_id: actor_id.to_owned(),
+                provider_id: provider_id.to_owned(),
+                link_name: self.config.provider_link_name.to_owned(),
+                contract_id: self.config.provider_contract_id.to_owned(),
+                values: self.config.values.to_owned(),
+                model_name: self.config.model_name.to_owned(),
+            })])
+        } else {
+            Ok(Vec::new())
+        }
+    }
+
+    async fn cleanup(&self) -> Result<Vec<Command>> {
+        if let (Ok(actor_id), Ok(provider_id)) = (self.actor_id().await, self.provider_id().await) {
+            Ok(vec![Command::DeleteLinkdef(DeleteLinkdef {
+                actor_id: actor_id.to_owned(),
+                provider_id: provider_id.to_owned(),
+                contract_id: self.config.provider_contract_id.to_owned(),
+                link_name: self.config.provider_link_name.to_owned(),
+                model_name: self.config.model_name.to_owned(),
+            })])
+        } else {
+            // If we never knew the actor/provider ID, this link scaler
+            // never created the link
+            Ok(Vec::new())
+        }
+    }
+
+    async fn backoff(&self, notifier: Sender<String>) {
+        let _ = notifier.send(String::with_capacity(0)).await;
+    }
+}
+
+impl<S: ReadStore + Send + Sync> LinkScaler<S> {
+    /// Construct a new LinkScaler with specified configuration values
+    #[allow(clippy::too_many_arguments)]
+    pub fn new(
+        store: S,
+        actor_reference: String,
+        provider_reference: String,
+        provider_contract_id: String,
+        provider_link_name: Option<String>,
+        lattice_id: String,
+        model_name: String,
+        values: HashMap<String, String>,
+    ) -> Self {
+        Self {
+            store,
+            actor_id: OnceCell::new(),
+            provider_id: OnceCell::new(),
+            config: LinkScalerConfig {
+                actor_reference,
+                provider_reference,
+                provider_contract_id,
+                provider_link_name: provider_link_name
+                    .unwrap_or_else(|| DEFAULT_LINK_NAME.to_string()),
+                lattice_id,
+                model_name,
+                values,
+            },
+        }
+    }
+
+    /// Helper function to retrieve the actor ID for the configured actor
+    async fn actor_id(&self) -> Result<&str> {
```

Comment:
Future improvement idea: Create a `ProviderRef` and `ActorRef` type that can be passed in instead so we don't have to copy paste this code everywhere